exercises/resistor-color-trio/test/Tests.hs

```diff
@@ -0,0 +1,106 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE NumDecimals #-}
+
+import Data.Foldable     (for_)
+import Test.Hspec        (Spec, describe, it, shouldBe)
+import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
+import Data.String       (fromString)
+
+import ResistorColors (Color(..), Resistor(..), label, ohms)
+
+main :: IO ()
+main = hspecWith defaultConfig {configFastFail = True} specs
+
+specs :: Spec
+specs = do
+  describe "label" $ for_ cases testLabel
+  describe "ohms" $ for_ cases testOhms
+  where
+    testLabel Case{..} = it (description input) $
+      label input `shouldBe` fromString expectedLabel
+      where
+        description Resistor{..} =
+          show bands ++ " should give " ++ expectedLabel
+
+    testOhms Case{..} = it (description input) $
+      ohms input `shouldBe` expectedOhms
+      where
+        description Resistor{..} =
+          show bands ++ " should give " ++ show expectedOhms
+
+data Case = Case { input         :: Resistor
+                 , expectedLabel :: String
+                 , expectedOhms  :: Int
+                 }
+
+cases :: [Case]
+cases = [ Case { input         = Resistor (Black, Black, Black)
+               , expectedLabel = "0 ohms"
+               , expectedOhms  = 0
+               }
+
+        , Case { input         = Resistor (Black, Yellow, Black)
+               , expectedLabel = "4 ohms"
+               , expectedOhms  = 4
+               }
+        , Case { input         = Resistor (Grey, Orange, Black)
+               , expectedLabel = "83 ohms"
+               , expectedOhms  = 83
+               }
+        , Case { input         = Resistor (Blue, Grey, Brown)
+               , expectedLabel = "680 ohms"
+               , expectedOhms  = 680
+               }
+        , Case { input         = Resistor (Red, Black, Red)
+               , expectedLabel = "2 kiloohms"
+               , expectedOhms  = 2e3
+               }
+        , Case { input         = Resistor (Green, Brown, Orange)
+               , expectedLabel = "51 kiloohms"
+               , expectedOhms  = 51e3
+               }
+        , Case { input         = Resistor (White, Violet, Yellow)
+               , expectedLabel = "970 kiloohms"
+               , expectedOhms  = 970e3
+               }
+
```

Comment:
~~I see that there is a newline here, perhaps because of the change in SI prefix between kilo and mega. I wondered why there is not a newline between (no prefix) and kilo, and no newline between mega and giga?~~

Nope, I was wrong because I didn't read the commit message, which was easily fixed by my reading the commit message. I didn't even think of the explanation in the commit message because there are five cases in the canonical data and six cases between the newlines here. I have no change to suggest. It is indeed useful to have every colour represented